pysatModels/utils/match.py

```diff
@@ -0,0 +1,279 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (C) 2022, pysat development team
+# Full license can be found in License.md
+# -----------------------------------------------------------------------------
+"""Routines to match modelled and observational data."""
+
+import datetime as dt
+import numpy as np
+import pandas as pds
+
+import pysat
+
+import pysatModels
+from pysatModels.utils.convert import load_model_xarray
+from pysatModels.utils import extract
+
+
+def collect_inst_model_pairs(start, stop, tinc, inst, inst_download_kwargs=None,
+                             model_load_rout=load_model_xarray,
+                             model_load_kwargs=None, inst_clean_rout=None,
+                             inst_lon_name=None, mod_lon_name=None,
+                             lon_pos='end', inst_name=None, mod_name=None,
+                             mod_datetime_name=None, mod_time_name=None,
+                             mod_units=None, sel_name=None, time_method='min',
+                             pair_method='closest', method='linear',
+                             model_label='model', comp_clean='clean'):
+    """Pair instrument and model data.
+
+    Parameters
+    ----------
+    start : dt.datetime
+        Starting datetime
+    stop : dt.datetime
+        Ending datetime
+    tinc : dt.timedelta
+        Time incriment for model files
+    inst : pysat.Instrument
+        Instrument object for which modelled data will be extracted
+    inst_download_kwargs : dict or NoneType
+        Optional keyword arguments for downloading instrument data
+        (default=None)
+    model_load_rout : func
+        Routine to load model data into an xarray using datetime as argument
+        input input and other necessary data as keyword arguments.  If the
+        routine requires a time-dependent filename, ensure that the load
+        routine uses the datetime input to construct the correct filename, as
+        done in load_model_xarray. (default=load_model_xarray)
+    model_load_kwargs : dict or NoneType
+        Keyword arguments for the model loading routine. (default=None)
+    inst_clean_rout : func
+        Routine to clean the instrument data
```

Comment:
Other keyword arguments in this docstring as well.
```suggestion
        Routine to clean the instrument data. (default=None)
```